## `nixpkgs-review` result

Generated using [`nixpkgs-review`](https://github.com/Mic92/nixpkgs-review).

Command: `nixpkgs-review pr 508520`
Commit: `57d5bcdbe858f16d33a45a6342966a0344863282`

---
### `x86_64-linux`
<details>
  <summary>:x: 1 package failed to build:</summary>
  <ul>
    <li>podman-desktop</li>
  </ul>
</details>

---
### `aarch64-linux`
<details>
  <summary>:x: 1 package failed to build:</summary>
  <ul>
    <li>podman-desktop</li>
  </ul>
</details>

---
### `aarch64-darwin`
<details>
  <summary>:x: 1 package failed to build:</summary>
  <ul>
    <li>podman-desktop</li>
  </ul>
</details>

---
<details>
<summary>Error logs: `x86_64-linux`</summary>
<details>
<summary>podman-desktop</summary>
<pre>/build /build/source
/build/source
Installing dependencies
 ERR_PNPM_UNSUPPORTED_ENGINE  Unsupported environment (bad pnpm and/or Node.js version)

Your Node version is incompatible with &quot;/build/source&quot;.

Expected version: &gt;=24.0.0
Got: v22.22.2

This is happening because the package&#x27;s manifest has an engines.node field specified.
To fix this issue, install the required Node version.

ERROR: pnpm failed to install dependencies

If you see ERR_PNPM_NO_OFFLINE_TARBALL above this, follow these to fix the issue:
1. Set pnpmDeps.hash to &quot;&quot; (empty string)
2. Build the derivation and wait for it to fail with a hash mismatch
3. Copy the &#x27;got: sha256-&#x27; value back into the pnpmDeps.hash field
</pre>
</details>
</details>

---
<details>
<summary>Error logs: `aarch64-linux`</summary>
<details>
<summary>podman-desktop</summary>
<pre>/build /build/source
/build/source
Installing dependencies
 ERR_PNPM_UNSUPPORTED_ENGINE  Unsupported environment (bad pnpm and/or Node.js version)

Your Node version is incompatible with &quot;/build/source&quot;.

Expected version: &gt;=24.0.0
Got: v22.22.2

This is happening because the package&#x27;s manifest has an engines.node field specified.
To fix this issue, install the required Node version.

ERROR: pnpm failed to install dependencies

If you see ERR_PNPM_NO_OFFLINE_TARBALL above this, follow these to fix the issue:
1. Set pnpmDeps.hash to &quot;&quot; (empty string)
2. Build the derivation and wait for it to fail with a hash mismatch
3. Copy the &#x27;got: sha256-&#x27; value back into the pnpmDeps.hash field
</pre>
</details>
</details>

---
<details>
<summary>Error logs: `aarch64-darwin`</summary>
<details>
<summary>podman-desktop</summary>
<pre>/nix/var/nix/builds/nix-52944-33610327 /nix/var/nix/builds/nix-52944-33610327/source
/nix/var/nix/builds/nix-52944-33610327/source
Installing dependencies
 ERR_PNPM_UNSUPPORTED_ENGINE  Unsupported environment (bad pnpm and/or Node.js version)

Your Node version is incompatible with &quot;/nix/var/nix/builds/nix-52944-33610327/source&quot;.

Expected version: &gt;=24.0.0
Got: v22.22.2

This is happening because the package&#x27;s manifest has an engines.node field specified.
To fix this issue, install the required Node version.

ERROR: pnpm failed to install dependencies

If you see ERR_PNPM_NO_OFFLINE_TARBALL above this, follow these to fix the issue:
1. Set pnpmDeps.hash to &quot;&quot; (empty string)
2. Build the derivation and wait for it to fail with a hash mismatch
3. Copy the &#x27;got: sha256-&#x27; value back into the pnpmDeps.hash field
</pre>
</details>
</details>

--- a/pkgs/by-name/po/podman-desktop/package.nix
+++ b/pkgs/by-name/po/podman-desktop/package.nix
@@ -5,7 +5,7 @@
   makeBinaryWrapper,
   copyDesktopItems,
   electron_39,
-  nodejs,
+  nodejs_24,
   pnpm_10,
   fetchPnpmDeps,
   pnpmConfigHook,
@@ -21,6 +21,8 @@
 }:
 
 let
+  nodejs = nodejs_24;
+  pnpm = pnpm_10.override { inherit nodejs; };
   electron = electron_39;
   appName = "Podman Desktop";
 in
@@ -64,7 +66,7 @@ stdenv.mkDerivation (finalAttrs: {
 
   pnpmDeps = fetchPnpmDeps {
     inherit (finalAttrs) pname version src;
-    pnpm = pnpm_10;
+    inherit pnpm;
     fetcherVersion = 2;
     hash = "sha256-nBjAmXzjR0qGCM91UAonQKP0NG7+DXImueSbhbnMK/k=";
   };
@@ -79,8 +81,8 @@ stdenv.mkDerivation (finalAttrs: {
   nativeBuildInputs = [
     makeBinaryWrapper
     nodejs
+    pnpm
     pnpmConfigHook
-    pnpm_10
   ]
   ++ lib.optionals (!stdenv.hostPlatform.isDarwin) [
     copyDesktopItems

--- a/pkgs/by-name/po/podman-desktop/package.nix
+++ b/pkgs/by-name/po/podman-desktop/package.nix
@@ -4,7 +4,7 @@
   fetchFromGitHub,
   makeBinaryWrapper,
   copyDesktopItems,
-  electron_39,
+  electron_41,
   nodejs_24,
   pnpm_10,
   fetchPnpmDeps,
@@ -23,12 +23,12 @@
 let
   nodejs = nodejs_24;
   pnpm = pnpm_10.override { inherit nodejs; };
-  electron = electron_39;
+  electron = electron_41;
   appName = "Podman Desktop";
 in
 stdenv.mkDerivation (finalAttrs: {
   pname = "podman-desktop";
-  version = "1.23.1";
+  version = "1.26.2";
 
   passthru.updateScript = _experimental-update-script-combinators.sequence [
     (nix-update-script { })
@@ -61,14 +61,14 @@ stdenv.mkDerivation (finalAttrs: {
     owner = "containers";
     repo = "podman-desktop";
     tag = "v${finalAttrs.version}";
-    hash = "sha256-08boCPsuT09OileZUWhB8awXWHrlJzoER2Bx0WXeOHU=";
+    hash = "sha256-VVyKC1z7YECZlbTaFaq2OwGg0k22qBbn/HEOYiJ8fcw=";
   };
 
   pnpmDeps = fetchPnpmDeps {
     inherit (finalAttrs) pname version src;
     inherit pnpm;
     fetcherVersion = 2;
-    hash = "sha256-nBjAmXzjR0qGCM91UAonQKP0NG7+DXImueSbhbnMK/k=";
+    hash = "sha256-tCp5qLZVo93H8VIToU3mkmwNsVXOAd1IEsL6RlazPXo=";
   };
 
   patches = [
@@ -76,7 +76,10 @@ stdenv.mkDerivation (finalAttrs: {
     ./extension-no-download-podman.patch
   ];
 
-  env.ELECTRON_SKIP_BINARY_DOWNLOAD = "1";
+  env = {
+    ELECTRON_SKIP_BINARY_DOWNLOAD = "1";
+    ELECTRON_OVERRIDE_DIST_PATH = electron.dist;
+  };
 
   nativeBuildInputs = [
     makeBinaryWrapper

